File: main/src/main/java/com/sedmelluq/discord/lavaplayer/track/playback/AudioFrameConsumer.java
package com.sedmelluq.discord.lavaplayer.track.playback;

/**
 * A consumer for audio frames
 */
public interface AudioFrameConsumer {
  /**
   * Consumes the frame, may block
   * @param frame The frame to consume
   * @throws InterruptedException When interrupted externally (or for seek/stop).
   */
  void consume(AudioFrame frame) throws InterruptedException;

  /**
   * Rebuild all caches frames
   * @param rebuilder The rebuilder to use
   */
  void rebuild(AudioFrameRebuilder rebuilder);
}

File: main/src/main/java/lavaplayer/track/playback/AudioFrameConsumer.kt
package lavaplayer.track.playback

/**
 * A consumer for audio frames
 */
interface AudioFrameConsumer {
    /**
     * Consumes the frame, may block
     *
     * @param frame The frame to consume
     * @throws InterruptedException When interrupted externally (or for seek/stop).
     */
    @Throws(InterruptedException::class)
    fun consume(frame: AudioFrame)

    /**
     * Rebuild all caches frames
     *
     * @param rebuilder The rebuilder to use
     */
    fun rebuild(rebuilder: AudioFrameRebuilder)
}
